peekingduck/pipeline/nodes/node.py

```diff
@@ -16,25 +16,38 @@
 Abstract Node class for all nodes
 """
 
+import pathlib
 import logging
+import collections
 from abc import ABCMeta, abstractmethod
 from typing import Any, Dict, List
 
+from peekingduck.configloader import ConfigLoader
+
 
 class AbstractNode(metaclass=ABCMeta):
     """
     Abstract Node class for inheritance by nodes.
     It defines default attributes and methods of a node.
     """
 
-    def __init__(self, config: Dict[str, Any], node_path: str) -> None:
+    def __init__(self, config: Dict[str, Any] = None,
+                 node_path: str = "", pkdbasedir: str = None, **kwargs: Any) -> None:
```

Comment:
`pkdbasedir: str = None` is an optional argument and is not needed in the nodes peekingduck provides (it is used for custom nodes). I think it may be okay for different nodes to have different number of arguments.